packages/client/hmi-client/src/composables/useChartSettings.ts

```diff
@@ -101,6 +101,24 @@ export function useChartSettings(
 		});
 	};
 
+	const updateChartPrimaryColor = (settings: ChartSetting, color: string) => {
+		const index = chartSettings.value.findIndex(
```

Comment:
You should be able to find the chart setting by it's id. e.g. `settings.id === chart.id`